src/plugins/gui/RewardsCardPlugin.tsx

```diff
@@ -285,13 +285,25 @@ export const makeRewardsCardPlugin: FiatPluginFactory = async params => {
     pluginId,
     startPlugin: async (startParams: FiatPluginStartParams) => {
       // Auth User:
-      const isAuthenticated = await provider.otherMethods.authenticate()
+      const startPluginInner = async (): Promise<RewardsCardItem[]> => {
+        const isAuthenticated = await provider.otherMethods.authenticate().catch(e => {
+          throw new Error(lstrings.rewards_card_error_authenticate)
+        })
 
-      // Get/refresh rewards cards:
-      if (isAuthenticated) {
-        rewardCards = await getRewardCards()
+        // Get/refresh rewards cards:
+        if (isAuthenticated) {
+          return await getRewardCards().catch(e => {
+            throw new Error(lstrings.rewards_card_error_retrieving_cards)
+          })
+        }
+        return rewardCards
       }
 
+      rewardCards = await showUi.showToastSpinner(
+        lstrings.loading,
+        runWithTimeout(startPluginInner(), 8000, new Error(lstrings.rewards_card_error_timeout_loading))
```

Comment:
Because we know that the API is low, can we increase from 8 seconds to 30, or 15 seconds? Ideally, the API gets faster, but I'd rather give it a bit more room to succeed. What do you think?